cli/internal/action/exportDocumentation.go

```diff
@@ -0,0 +1,266 @@
+package action
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"hyaline/internal/config"
+	"hyaline/internal/docs"
+	"hyaline/internal/sqlite"
+	"log/slog"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+)
+
+type ExportDocumentationArgs struct {
+	Documentation string
+	Format        string
+	Includes      []string
+	Excludes      []string
+	Output        string
+}
+
+type ExportFormatType string
+
+func (t ExportFormatType) String() string {
+	return string(t)
+}
+
+func (t ExportFormatType) IsValid() bool {
+	switch t {
+	case ExportFormatFs, ExportFormatLlmsFullTxt, ExportFormatJson, ExportFormatSqlite:
+		return true
+	default:
+		return false
+	}
+}
+
+func (t ExportFormatType) PossibleValues() string {
+	return fmt.Sprintf("%s, %s, %s, %s", ExportFormatFs, ExportFormatLlmsFullTxt, ExportFormatJson, ExportFormatSqlite)
+}
+
+const (
+	ExportFormatFs          ExportFormatType = "fs"
+	ExportFormatLlmsFullTxt ExportFormatType = "llmsfulltxt"
+	ExportFormatJson        ExportFormatType = "json"
+	ExportFormatSqlite      ExportFormatType = "sqlite"
+)
+
+func ExportDocumentation(args *ExportDocumentationArgs) error {
+	slog.Info("Exporting Documentation",
+		"documentation", args.Documentation,
+		"format", args.Format,
+		"includes", args.Includes,
+		"excludes", args.Excludes,
+		"output", args.Output,
+	)
+
+	// Validate format
+	format := ExportFormatType(args.Format)
+	if !format.IsValid() {
+		slog.Debug("action.ExportDocumentation received an invalid format")
+		return fmt.Errorf("invalid format, got: %s, wanted one of: %s", format.String(), format.PossibleValues())
+	}
+
+	// Parse and validate includes/excludes
+	includes := []*docs.DocumentURI{}
+	excludes := []*docs.DocumentURI{}
+	for _, include := range args.Includes {
+		uri, err := docs.NewDocumentURI(include)
+		if err != nil {
+			slog.Debug("action.ExportDocumentation could not parse include", "include", include)
+			return err
+		}
+		includes = append(includes, uri)
+	}
+	for _, exclude := range args.Excludes {
+		uri, err := docs.NewDocumentURI(exclude)
+		if err != nil {
+			slog.Debug("action.ExportDocumentation could not parse exclude", "exclude", exclude)
+			return err
+		}
+		excludes = append(excludes, uri)
+	}
+
+	// Ensure output path does not exist
+	outputAbsPath, err := filepath.Abs(args.Output)
+	if err != nil {
+		slog.Debug("action.ExportDocumentation could not get an absolute path for output", "output", args.Output, "error", err)
+		return err
+	}
+	_, err = os.Stat(outputAbsPath)
+	if err == nil {
+		slog.Debug("action.ExportDocumentation detected that output already exists", "absPath", outputAbsPath)
+		return errors.New("output file already exists")
```

Comment:
nit: (since it's not always a file)
```suggestion
		return errors.New("output path already exists")
```